### PR TITLE
Add support for Python 3.12

### PR DESCRIPTION
**Pull request by robsdedude**
Depends on:
 * https://github.com/neo-technology/driver-teamcity-pipelines/pull/206 (then re-run the pipelines)

### CODE REVIEW

**Review by AndyHeap-NeoTech — APPROVED** [review]
All makes sense to me.

File: src/neo4j/_async_compat/shims/__init__.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 # we remove support for Python 3.9.
 
 
-if sys.version_info >= (3, 8):
+if (3, 12) > sys.version_info >= (3, 8):
```
Comment (thelonelyvulpes) [inline src/neo4j/_async_compat/shims/__init__.py:32]: Could include the note about why this shim isn't need for 3.12 onwards.

File: pyproject.toml [hunk 2]
```diff
@@ -50,17 +50,19 @@ pandas = [
     "pandas >= 1.1.0, < 3.0.0",
     "numpy >= 1.7.0, < 2.0.0",
 ]
+pyarrow = ["pyarrow >= 1.0.0"]
```
Comment (thelonelyvulpes) [inline pyproject.toml:54]: I believe this file is missing the classifier.
```
    "Programming Language :: Python :: 3.12",
```

**Review by thelonelyvulpes — APPROVED** [review]
🐍

File: src/neo4j/_async_compat/shims/__init__.py (hunk 1 above)
Comment (thelonelyvulpes) [reply src/neo4j/_async_compat/shims/__init__.py:32]:
```suggestion
# wait_for is only broken in Python 3.8-3.11 (inclusive)
if (3, 12) > sys.version_info >= (3, 8):
```

**Pull request merged**